includes/admin/class-wc-stripe-settings-controller.php

```diff
@@ -140,50 +140,29 @@ private function is_payments_onboarding_task_completed(): bool {
 	}
 
 	/**
-	 * AJAX handler to get OAuth URLs for the configuration modal
+	 * AJAX handler to generate OAuth URL on-demand
 	 */
-	public function ajax_get_oauth_urls() {
+	public function ajax_get_oauth_url() {
 		// Check nonce and capabilities
-		if ( ! check_ajax_referer( 'wc_stripe_get_oauth_urls', 'nonce', false ) ||
+		if ( ! check_ajax_referer( 'wc_stripe_get_oauth_url', 'nonce', false ) ||
 			! current_user_can( 'manage_woocommerce' ) ) {
 			wp_send_json_error( [ 'message' => __( 'You do not have permission to do this.', 'woocommerce-gateway-stripe' ) ] );
 			return;
 		}
 
-		$oauth_url      = woocommerce_gateway_stripe()->connect->get_oauth_url();
-		$test_oauth_url = woocommerce_gateway_stripe()->connect->get_oauth_url( '', 'test' );
-
-		wp_send_json_success(
-			[
-				'oauth_url'      => is_wp_error( $oauth_url ) ? '' : $oauth_url,
-				'test_oauth_url' => is_wp_error( $test_oauth_url ) ? '' : $test_oauth_url,
-			]
-		);
-	}
-
-	/**
-	 * Determines if OAuth URLs need to be generated.
-	 * URLs are needed for new accounts or accounts not connected via OAuth.
-	 *
-	 * @return bool True if OAuth URLs are needed
-	 */
-	public function needs_oauth_urls() {
-		$settings      = WC_Stripe_Helper::get_stripe_settings();
-		$has_live_keys = ! empty( $settings['publishable_key'] ) && ! empty( $settings['secret_key'] );
-		$has_test_keys = ! empty( $settings['test_publishable_key'] ) && ! empty( $settings['test_secret_key'] );
-
-		// If no keys at all, we need OAuth URLs for new account setup
-		if ( ! $has_live_keys && ! $has_test_keys ) {
-			return true;
+		$mode = isset( $_POST['mode'] ) ? sanitize_text_field( wp_unslash( $_POST['mode'] ) ) : 'live';
+		if ( ! in_array( $mode, [ 'live', 'test' ], true ) ) {
+			$mode = 'live';
```

Comment:
Should we not default to test mode when we have a caller behaving poorly? It feels like that should result be less problematic by default for cases where we get a nonce leaked etc.